lib/active_merchant/billing/gateways/ct_payment.rb

```diff
@@ -0,0 +1,297 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class CtPaymentGateway < Gateway
+      self.test_url = 'https://test.ctpaiement.ca/v1/'
+      self.live_url = 'https://example.com/live'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'http://www.example.net/'
+      self.display_name = 'CT Payment'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        '14' => STANDARD_ERROR_CODE[:invalid_number],
+        '05' => STANDARD_ERROR_CODE[:card_declined],
+        'M6' => STANDARD_ERROR_CODE[:card_declined],
+        '9068' => STANDARD_ERROR_CODE[:incorrect_number],
+        '9067' => STANDARD_ERROR_CODE[:incorrect_number]
+      }
+      CARD_BRAND = {
+        'american_express' => 'A',
+        'master' => 'M',
+        'diners_club' => 'I',
+        'visa' => 'V',
+        'discover' => 'O'
+      }
+
+      PARAMETERS_ORDER = [
+        :CompanyNumber,
+        :MerchantNumber,
+        :CustomerNumber,
+        :Amount,
+        :InvoiceNumber,
+        :InputType,
+        # :Name,
+        # :Email,
+        :CardType,
+        :CardNumber,
+        :ExpirationDate,
+        :MerchantTerminalNumber,
+        :LanguageCode,
+        :CurrencyCode,
+        :OperatorID
+      ]
+      def initialize(options={})
+        requires!(options, :api_key, :company_number, :merchant_number)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        add_terminal_number(post, options)
+        add_money(post, money)
+        add_operator_id(post, options)
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+        response = if payment.is_a?(String)
```

Comment:
You _might_ be able to fit this on one line via ternary, but your call; I find this very readable as-is.

Reply:
@bpollack yea this is a remanent of me getting stuck in rubocop looks. It was complaining that the one line ternary was too long so then I did multiline then it complained that it should be a if else statement. Going back and fixing that stuff now.